ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/BeaconState.java

```diff
@@ -13,83 +13,57 @@
-import static java.lang.Math.toIntExact;
-import static tech.pegasys.artemis.datastructures.Constants.DOMAIN_DEPOSIT;
-import static tech.pegasys.artemis.datastructures.Constants.EPOCH_LENGTH;
-import static tech.pegasys.artemis.datastructures.Constants.GENESIS_EPOCH;
-import static tech.pegasys.artemis.datastructures.Constants.GENESIS_SLOT;
-import static tech.pegasys.artemis.datastructures.Constants.SHARD_COUNT;
-import static tech.pegasys.artemis.util.bls.BLSVerify.bls_verify;
-
-import com.google.common.annotations.VisibleForTesting;
 import com.google.common.primitives.UnsignedLong;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
 import net.consensys.cava.bytes.Bytes32;
 import net.consensys.cava.bytes.Bytes48;
-import tech.pegasys.artemis.datastructures.Constants;
 import tech.pegasys.artemis.datastructures.blocks.Eth1Data;
 import tech.pegasys.artemis.datastructures.blocks.Eth1DataVote;
-import tech.pegasys.artemis.datastructures.operations.AttestationData;
-import tech.pegasys.artemis.datastructures.operations.Deposit;
-import tech.pegasys.artemis.datastructures.operations.DepositInput;
-import tech.pegasys.artemis.datastructures.state.CrosslinkCommittee;
 import tech.pegasys.artemis.datastructures.state.CrosslinkRecord;
 import tech.pegasys.artemis.datastructures.state.Fork;
 import tech.pegasys.artemis.datastructures.state.PendingAttestationRecord;
-import tech.pegasys.artemis.datastructures.state.Validator;
 import tech.pegasys.artemis.datastructures.state.Validators;
-import tech.pegasys.artemis.statetransition.util.BeaconStateUtil;
-import tech.pegasys.artemis.statetransition.util.TreeHashUtil;
 
 public class BeaconState {
-
   // Misc
-  private long slot;
-  private long genesis_time;
-  private Fork fork;
+  private UnsignedLong slot;
+  private UnsignedLong genesis_time;
+  private Fork fork; // For versioning hard forks
 
   // Validator registry
   private Validators validator_registry;
-  private ArrayList<Double> validator_balances;
-  private long validator_registry_update_epoch;
+  private ArrayList<UnsignedLong> validator_balances;
+  private UnsignedLong validator_registry_update_epoch;
 
   // Randomness and committees
   private ArrayList<Bytes32> latest_randao_mixes;
-  private long previous_epoch_start_shard;
-  private long current_epoch_start_shard;
-  private long previous_calculation_epoch;
-  private long current_calculation_epoch;
-  private Bytes32 previous_epoch_seed;
-  private Bytes32 current_epoch_seed;
+  private UnsignedLong previous_epoch_start_shard;
+  private UnsignedLong current_epoch_start_shard;
+  private UnsignedLong previous_calculation_epoch;
+  private UnsignedLong current_calculation_epoch;
 
   // Finality
-  private long previous_justified_slot;
-  private long justified_slot;
-  private long justification_bitfield;
-  private long finalized_slot;
+  private Bytes32 previous_epoch_seed;
```

Comment:
previous_epoch_seed & current_epoch_seed is under //Randomness and committees in the v0.01 spec

Reply:
Good catch. Updated